packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts

```diff
@@ -0,0 +1,323 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { assert, fromBase64ToUtf8, hashFile, IsoBuffer, Uint8ArrayToString } from "@fluidframework/common-utils";
+import { ISummaryContext } from "@fluidframework/driver-definitions";
+import { getGitType } from "@fluidframework/protocol-base";
+import * as api from "@fluidframework/protocol-definitions";
+import { PerformanceEvent } from "@fluidframework/telemetry-utils";
+import {
+    IBlob,
+    ISnapshotRequest,
+    ISnapshotResponse,
+    ISnapshotTree,
+    ISnapshotTreeBaseEntry,
+    SnapshotTreeEntry,
+    SnapshotTreeValue,
+    SnapshotType,
+} from "./contracts";
+import { EpochTracker, FetchType } from "./epochTracker";
+import { getUrlAndHeadersWithAuth } from "./getUrlAndHeadersWithAuth";
+import { getWithRetryForTokenRefresh } from "./odspUtils";
+import { TokenFetchOptions } from "./tokenFetch";
+
+/* eslint-disable max-len */
+
+interface IDedupCaches {
+    blobShaToPath: Map<string, string>,
+    pathToBlobSha: Map<string, string>,
+    treesPathToTree: Map<string, api.ISummaryTree>,
+}
+
+export class OdspSummaryUploadManager {
+    // This cache is associated with mapping sha to path for previous summary which belongs to last summary handle.
+    private blobTreeDedupCaches: IDedupCaches = {
+        blobShaToPath: new Map(),
+        pathToBlobSha: new Map(),
+        treesPathToTree: new Map(),
+    };
+
+    private backupBlobTreeDedupCaches: IDedupCaches = {
```

Comment:
I'm not sure "backup" is the right word. "Previous"? Or I guess it's N-2? Some name that more clearly indicates purpose would be nice

Reply:
Added comment and changed the name to previous.